xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-notifications/src/main/resources/templates/mentions/mention.vm

```diff
@@ -82,8 +82,9 @@
         #set ($descriptionKey = 'mentions.event.mention.description.comment')
       #elseif($location == "ANNOTATION")
         #set ($descriptionKey = 'mentions.event.mention.description.annotation')
-      #elseif($location == "AWM_FIELD")
-        #set ($descriptionKey = 'mentions.event.mention.description.awmField')
+      #elseif($location == "AWM_FIELD" || $location == "TEXT_FIELD")
+        ## TODO: AWM_FIELD is deprecated since 14.4.8, 14.10.6, and 15.2RC1
```

Comment:
Why a TODO? We'll need to support it anyway so not much to do...

Reply:
I need to rephrase it. The todo is mainly to remind me to check if `$location == "AWM_FIELD"` can be moved to legacy (but looks tricky).

Reply:
You would not have needed all those `ifs` if you had generated the translation keys with the location value directly instead of doing this mapping.

Reply:
At the time, it was not obvious that all sentences would have the same structure. But in hindsight, I agree with you.

Reply:
Now since the content of those keys is not great in terms of "translability" as reported by @michitux maybe it's a good excuse to refactor this logic and generate new keys (based on the location value this time).

Reply:
@tmortagne Actually I'm not 100% sure to understand what you mean by `generate new keys (based on the location value this time).` Do you mean using the `choice` feature of message format combined with the index of the enum values?

Reply:
I mean

```
"mentions.event.mention.description.${location}"
```